File: src/leetcode/2026/06/Solution989.cpp
#include <vector>
using namespace std;

class Solution989 {
public:
    vector<int> addToArrayForm(vector<int>& num, int k) {
        vector<int> result;
        int carry = 0;
        int n = num.size();
        for (int i = n - 1; i >= 0 || k > 0; --i) {
            if (i >= 0) {
                k += num[i];
            }
            result.push_back(k % 10);
            k /= 10;
        }
        reverse(result.begin(), result.end());
        return result;
    }
};

int main() {
    Solution989 solution;
    vector<int> num = {1, 2, 0, 0};
    int k = 34;
    vector<int> result = solution.addToArrayForm(num, k);
    for (int digit : result) {
        printf("%d ", digit);
    }
    printf("\n");

    vector<int> num2 = {2, 7, 4};
    int k2 = 181;
    vector<int> result2 = solution.addToArrayForm(num2, k2);
    for (int digit : result2) {
        printf("%d ", digit);
    }
    printf("\n");

    vector<int> num3 = {2, 1, 5};
    int k3 = 806;
    vector<int> result3 = solution.addToArrayForm(num3, k3);
    for (int digit : result3) {
        printf("%d ", digit);
    }
    printf("\n");

    return 0;
}

File: src/leetcode/2026/06/Solution989.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Solution989 {
    public List<Integer> addToArrayForm(int[] num, int k) {
        List<Integer> result = new ArrayList<>();
        int carry = 0;
        int n = num.length;
        for (int i = n - 1; i >= 0; i--) {
            int sum = num[i] + k % 10 + carry;
            carry = sum / 10;
            result.add(sum % 10);
            k /= 10;
        }
        while (k > 0) {
            int sum = k % 10 + carry;
            carry = sum / 10;
            result.add(sum % 10);
            k /= 10;
        }
        if (carry > 0) {
            result.add(carry);
        }
        Collections.reverse(result);
        return result;
    }

    public static void main(String[] args) {
        Solution989 solution = new Solution989();

        int[] num1 = { 1, 2, 0, 0 };
        int k1 = 34;
        List<Integer> result1 = solution.addToArrayForm(num1, k1);
        System.out.print("Output for num1 and k1: ");
        for (int digit : result1) {
            System.out.print(digit + " ");
        }
        System.out.println(); // Output: [1, 2, 3, 4]

        int[] num2 = { 2, 7, 4 };
        int k2 = 181;
        List<Integer> result2 = solution.addToArrayForm(num2, k2);
        System.out.print("Output for num2 and k2: ");
        for (int digit : result2) {
            System.out.print(digit + " ");
        }
        System.out.println(); // Output: [4, 5, 5]

        int[] num3 = { 2, 1, 5 };
        int k3 = 806;
        List<Integer> result3 = solution.addToArrayForm(num3, k3);
        System.out.print("Output for num3 and k3: ");
        for (int digit : result3) {
            System.out.print(digit + " ");
        }
        System.out.println(); // Output: [1, 0, 2, 1]
    }
}
